File: src/test/java/com/wire/xenon/DAOTest.java
package com.wire.xenon;

import com.wire.xenon.crypto.storage.IdentitiesDAO;
import com.wire.xenon.state.StatesDAO;
import org.junit.jupiter.api.Test;

import java.util.UUID;

public class DAOTest extends DatabaseTestBase {
    @Test
    public void testIdentitiesDAO() {
        final IdentitiesDAO identitiesDAO = jdbi.onDemand(IdentitiesDAO.class);
        final String id = UUID.randomUUID().toString();

        final int insert = identitiesDAO.insert(id, id.getBytes());
        final byte[] bytes = identitiesDAO.get(id).data;
        final int delete = identitiesDAO.delete(id);
    }

    @Test
    public void testStatesDAO() {
        final StatesDAO statesDAO = jdbi.onDemand(StatesDAO.class);
        final UUID id = UUID.randomUUID();
        final String text = "{\"some\" : \"text\"}";

        final int insert = statesDAO.insert(id, text);
        final String dbText = statesDAO.get(id);
        final int delete = statesDAO.delete(id);
    }
}

File: src/test/java/com/wire/kalium/DAOTest.kt
package com.wire.kalium

import java.util.UUID
import com.wire.kalium.state.StatesDAO
import com.wire.kalium.crypto.storage.IdentitiesDAO
import org.junit.jupiter.api.Test

class DAOTest : DatabaseTestBase() {
    @Test
    fun testIdentitiesDAO() {
        val identitiesDAO: IdentitiesDAO = DatabaseTestBase.Companion.jdbi.onDemand<IdentitiesDAO?>(IdentitiesDAO::class.java)
        val id = UUID.randomUUID().toString()
        val insert = identitiesDAO.insert(id, id.toByteArray())
        val bytes = identitiesDAO[id].data
        val delete = identitiesDAO.delete(id)
    }

    @Test
    fun testStatesDAO() {
        val statesDAO: StatesDAO = DatabaseTestBase.Companion.jdbi.onDemand<StatesDAO?>(StatesDAO::class.java)
        val id = UUID.randomUUID()
        val text = "{\"some\" : \"text\"}"
        val insert = statesDAO.insert(id, text)
        val dbText = statesDAO[id]
        val delete = statesDAO.delete(id)
    }
}
